app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -419,6 +443,12 @@ fun KeyboardScreen(
                                     },
                                     onSwitchLanguage = onSwitchLanguage,
                                     onSwitchPosition = onSwitchPosition,
+                                    secondaryKey = secondaryKeyboard?.arr?.getOrNull(i)?.getOrNull(j),
```

Comment:
This should be something like `if (capsLock) {keyboardDefinition.modes.main} else {keyboardDefinitionModes.shifted}?.arr?....` 


Then you don't have to rebuild these layouts again.